lib/liquid/traversal.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+module Liquid
+  class Traversal
+    def self.for(node, callbacks = Hash.new(proc {}))
+      kase = CASES.find { |(klass, _)| node.is_a?(klass) }&.last
+      (kase || self).new(node, callbacks)
+    end
+
+    def initialize(node, callbacks)
+      @node = node
+      @callbacks = callbacks
+    end
+
+    def callback_for(*classes, &block)
```

Comment:
I'm not super happy with this name, but can't think of any great alternatives. Some options I considered:

- `add_callback`
- `add_callback_for`
- `add_visitor`
- `add_visitor_for`
- `on_visit`
- `[]=`

WDYT?